src/dctkit/dec/cochain.py

```diff
@@ -394,3 +394,76 @@ def sym(c: Cochain) -> Cochain:
         its symmetric part.
     """
     return scalar_mul(add(c, transpose(c)), 0.5)
+
+
+def convolution(c: Cochain, kernel: Cochain, kernel_window: float) -> Cochain:
+    """ Compute the convolution between two scalar 0-cochains.
+
+    Args:
+        c: a scalar 0-cochain.
+        kernel: the scalar 0-cochain kernel.
+        kernel_window: the kernel window.
+
+    Returns:
+        the convolution rho*kernel.
+    """
+    # we build a kernel matrix K by rolling the kernel vector k + 1 times, where
+    # k is the kernel window. In this way we can express the
+    # convolution between c and k as SK @c_coeffs where S is the hodge star
+    n = len(c.coeffs)
+    K = jnp.zeros((n, n), dtype=dt.float_dtype)
+    buffer = jnp.empty((n, n*2 - 1))
+
+    # generate a wider array that we want a slice into
+    buffer = buffer.at[:, :n].set(kernel.coeffs[:n].T)
+    buffer = buffer.at[:, n:].set(kernel.coeffs[:n-1].T)
+
+    rolled = buffer.reshape(-1)[n-1:-1].reshape(n, -1)
+    K_full_roll = jnp.roll(rolled[:, :n], shift=1, axis=0)
+    K_non_zero = K_full_roll[:n - kernel_window + 1]
+    K = K.at[:n - kernel_window + 1, :].set(K_non_zero)
+    kernel_coch = Cochain(c.dim, c.is_primal, c.complex, K)
```

Comment:
We pass the argument "kernel" to the function. "Kernel" is a cochain. Here we define kernel_coch: what should we expect from this variable? Wasn't the kernel a cochain already? This part of the code is not clear: there are no comments and the names of the variables do not help understanding what it does.